models/user.go

```diff
@@ -1332,6 +1333,142 @@ func GetWatchedRepos(userID int64, private bool) ([]*Repository, error) {
 	return repos, nil
 }
 
+// existsInSlice returns true if string exists in slice
+func existsInSlice(target string, slice []string) bool {
+	sort.Strings(slice)
+	i := sort.Search(len(slice),
+		func(i int) bool { return slice[i] >= target })
+	if i < len(slice) && slice[i] == target {
+		return true
+	}
+	return false
+}
+
+// IsEqualSlice returns true if slices are equal
+func IsEqualSlice(target []string, source []string) bool {
+	sort.Strings(target)
+	sort.Strings(source)
+
+	if len(target) != len(source) {
+		return false
+	}
+
+	if (target == nil) != (source == nil) {
+		return false
+	}
+
+	for i, v := range target {
+		if v != source[i] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// deleteKeysMarkedForDeletion returns true if ssh keys needs update
+func deleteKeysMarkedForDeletion(keys []string) (sshKeysNeedUpdate bool, err error) {
+	// Start session
+	sess := x.NewSession()
+	defer sessionRelease(sess)
```

Comment:
Rebase from master is needed as `sessionRelease` function has been removed and this line must be replaced with `defer sess.Close()`